pkg/controller/oneagent/oneagent_update.go

```diff
@@ -207,3 +209,43 @@ func (r *ReconcileOneAgent) setVersionByIP(instance dynatracev1alpha1.BaseOneAge
 	}
 	return nil
 }
+
+func isDesiredNewer(actual string, desired string, logger logr.Logger) bool {
+	aa := strings.Split(actual, ".")
+	da := strings.Split(desired, ".")
+
+	for i := 0; i < len(aa); i++ {
+		if i == len(aa)-1 {
+			if aa[i] < da[i] {
+				return true
+			} else if aa[i] > da[i] {
+				logger.Error(nil, "downgrade detected! downgrades are not supported")
```

Comment:
Does this `nil` not break anything?

Reply:
Created a separate error for the downgrade case